web/pgadmin/tools/sqleditor/static/js/components/QueryToolComponent.jsx

```diff
@@ -487,6 +486,17 @@ export default function QueryToolComponent({params, pgWindow, pgAdmin, selectedN
     });
   };
 
+  useEffect(()=> {
+    // Add beforeunload event if "Confirm on close or refresh" option is enabled in the preferences.
+    if(qtState.preferences.browser.confirm_on_refresh_close){
```

Comment:
I think we'll need the else case as well. A user can change preference after query tool opens. So the already bound event will not be removed in that case.

Reply:
Done, also added the check-in ERD tool.